src/app-layer-htp-file.c

```diff
@@ -225,34 +228,108 @@ int HTPParseContentRange(bstr * rawvalue, HtpContentRange *range)
+        s->events++;
+
+        SCLogDebug("parsing range failed, going back to normal file");
+        return HTPFileOpen(
+                s, filename, (uint32_t)filename_len, data, data_len, txid, STREAM_TOCLIENT);
+    }
+    /* crparsed.end <= 0 means a range with only size
+     * this is the answer to an unsatisfied range with the whole file
+     * crparsed.size <= 0 means an unknown size, so we do not know
+     * when to close it...
+     */
+    if (crparsed.end <= 0 || crparsed.size <= 0) {
+        SCLogDebug("range without all information");
+        return HTPFileOpen(
+                s, filename, (uint32_t)filename_len, data, data_len, txid, STREAM_TOCLIENT);
+    } else if (crparsed.end == crparsed.size - 1 && crparsed.start == 0) {
+        SCLogDebug("range without all information");
+        return HTPFileOpen(
+                s, filename, (uint32_t)filename_len, data, data_len, txid, STREAM_TOCLIENT);
+    }
+
+    flags = FileFlowToFlags(s->f, STREAM_TOCLIENT);
+    if ((s->flags & HTP_FLAG_STORE_FILES_TS) ||
+            ((s->flags & HTP_FLAG_STORE_FILES_TX_TS) && txid == s->store_tx_id)) {
+        flags |= FILE_STORE;
+        flags &= ~FILE_NOSTORE;
+    } else if (!(flags & FILE_STORE) && (s->f->file_flags & FLOWFILE_NO_STORE_TC)) {
+        flags |= FILE_NOSTORE;
+    }
+
     FileContainer * files = s->files_tc;
     if (files == NULL) {
-        SCLogDebug("no files in state");
+        s->files_tc = FileContainerAlloc();
+        if (s->files_tc == NULL) {
+            SCReturnInt(-1);
+        }
+        files = s->files_tc;
+    }
+
+    if (FileOpenFileWithId(files, &s->cfg->response.sbcfg, s->file_track_id++, filename,
+                filename_len, data, data_len, flags) != 0) {
         SCReturnInt(-1);
     }
+    FileSetTx(files->tail, txid);
 
-    HtpContentRange crparsed;
-    if (HTPParseContentRange(rawvalue, &crparsed) != 0) {
-        SCLogDebug("parsing range failed");
-        SCReturnInt(-2);
+    if (FileSetRange(files, crparsed.start, crparsed.end) < 0) {
+        SCLogDebug("set range failed");
     }
-    if (crparsed.end <= 0) {
-        SCLogDebug("negative end in range");
-        SCReturnInt(-3);
+    htp_tx_t *tx = htp_list_get(s->conn->transactions, txid);
+    if (!tx) {
+        SCReturnInt(-1);
     }
-    int retval = FileSetRange(files, crparsed.start, crparsed.end);
-    if (retval == -1) {
-        SCLogDebug("set range failed");
+    uint8_t *keyurl;
+    size_t keylen;
+    if (tx->request_hostname != NULL) {
+        keylen = bstr_len(tx->request_hostname) + filename_len + 1;
+        keyurl = SCMalloc(keylen);
+        if (keyurl == NULL) {
+            SCReturnInt(-1);
+        }
+        memcpy(keyurl, bstr_ptr(tx->request_hostname), bstr_len(tx->request_hostname));
+        memcpy(keyurl + bstr_len(tx->request_hostname), filename, filename_len);
+        keyurl[keylen - 1] = 0;
+    } else {
+        // do not reassembly file without host info
+        SCReturnInt(-1);
```

Comment:
what does this mean and what should the consequences be?

Reply:
If we do not have the host name, we should fall back to `HTPFileOpen` (instead of returning -1, thanks)